daffodil-core/src/main/scala/org/apache/daffodil/grammar/ModelGroupGrammarMixin.scala

```diff
@@ -61,7 +61,9 @@ trait ModelGroupGrammarMixin
         // there are no local delimiters.
         //
         ) {
-        val content = initiatorRegion ~ groupContent ~ terminatorRegion
+        val content = initiatorRegion ~ (
+          if (turnOnSharing) sharedModelGroup.groupContent
+          else groupContent) ~ terminatorRegion
         self match {
```

Comment:
The above is 1/2 the magic here. If turnOnSharing, then the groupContent grammar term is taken from the shared object, not the unique object. Hence it, and the parser/unparser generated by it, are shared.